script.js

```diff
@@ -1,4 +1,200 @@
+// generate random dice number
+function rollDice() {
+  let randomNum = Math.floor(Math.random() * 6) + 1;
+  return randomNum;
+}
+// arrange the order of dices as per user input.
+function arrangeDice(userInput, firstDice, secondDice) {
+  let arrangedResult;
+  if (userInput == 'dice1') {
+    diceOrder = 'dice1';
+  }
+  else {
+    diceOrder = 'dice2';
+  }
+  if (diceOrder == 'dice1') {
+    arrangedResult = `${firstDice}${secondDice}`;
+  }
+  else {
+    arrangedResult = `${secondDice}${firstDice}`;
+  }
+  return Number(arrangedResult);
+}
+// function is used for finding winner with highest (normal) as well as lowest score modes.
+function findWinner(playersScoreArray, mode) {
+  if (mode == 'normal') {
+    return Math.max.apply(null, playersScoreArray);
+  }
+  return Math.min.apply(null, playersScoreArray);
+}
+function autoGenNumber(rolledNumArray) {
+  let array = rolledNumArray;
+  let finalNumber = 0;
+  let length = array.length;
+  for (let i = 0; array.length >= 1; i++) {
+    let max = Math.max.apply(null, array);
+    finalNumber = finalNumber * 10 + max;
+    rolledNumArray = array.splice(array.indexOf(max), 1);
+  }
+  return finalNumber;
+}
+function setMode(userInput) {
+  let mode;
+  if (userInput == 'manual') {
+    mode = 'manual';
+  }
+  else if (userInput == 'auto') {
+    mode = 'auto';
+  }
+  return mode;
+}
+function iterateNumOfPlayers(totalPlayers, mode, userInput) {
```

Comment:
Lots of comments needed and missing from this function. Why do you perform each section of logic? Can be explained in comments.